add --passin --passout for alternative pass src/dst

luizluca opened the pull request:
Batch operations cannot be automated if openssl keeps asking
for a password. These new options allow the user to specify
a new source for password, using any openssl password options
like pass:1234 or env:var

Aborts build-ca if privkey generation fails.

Signed-off-by: Luiz Angelo Daros de Luca <luizluca@gmail.com>

This is based on #240 (pending)
This fixes #103, #101, #73, #18

VVD commented:
I'm using this hack:
```
--- easyrsa.real.orig
+++ easyrsa.real
@@ -675,6 +675,9 @@
        # Support batch by internal caller:
        [ "$3" = "batch" ] && EASYRSA_BATCH=1
 
+       # HACK: Support password for ca:
+       [ -n "$4" ] && opts="$opts -passin pass:$4"
+
        verify_ca_init
 
        # Check argument sanity:
```

luizluca commented:
Rebased, signed and fixed a minor bug (introduced by shellcheck suggestion).
FYI, this PR is based on #283.

op_test.sh could get a lot of new tests with passwords

luizluca commented:
Rebased on #240 , retarget to master and PGP signed

jkroepke commented:
@ecrist Any chance to see this in a released version?
